gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/MRCompactorAvroKeyDedupJobRunner.java

```diff
@@ -99,7 +108,9 @@ protected void configureJob(Job job) throws IOException {
 
   private void configureSchema(Job job) throws IOException {
     Schema newestSchema = getNewestSchemaFromSource(job);
-    AvroJob.setInputKeySchema(job, newestSchema);
+    if (useSingleInputSchema) {
+      AvroJob.setInputKeySchema(job, newestSchema);
+    }
     AvroJob.setMapOutputKeySchema(job, this.shouldDeduplicate ? getKeySchema(job, newestSchema) : newestSchema);
```

Comment:
I added the change in the `AvroKeyMapper (`keySchema = ((AvroCombineFileSplit) context.getInputSplit()).getSchema();`) to my commit